configure checks compiling test code should include stdlib.h

from config.log
```
configure:4969: checking whether long long int is 64 bits
[cross-compile branch is taken]
configure:5027: i486-linux-musl-gcc -L/home/sabotage/www/building/lib -isystem ...
conftest.c:58:1: warning: return type defaults to 'int' [-Wreturn-type]
conftest.c: In function 'main':
conftest.c:59:3: warning: implicit declaration of function 'exit' [-Wimplicit-function-declaration]
conftest.c:59:3: warning: incompatible implicit declaration of built-in function 'exit' [enabled by default]
```
exit() is declared by stdlib.h, but on GLIBC it somehow gets magically pulled in when including stdio.h, which is a namespace violation. on musl not, and so this warning is raised.
that would mean if i have -Werror-implicit-function-declaration in my CFLAGS, the configure check would fail because of that warning and the result would be misinterpreted, either giving me a broken build or not even letting me get there.

unfortunately i was not able to find where in configure.ac the code to compile in the cross-compile scenario comes from, so i couldnt make a PR.

--- a/acinclude.m4
+++ b/acinclude.m4
@@ -145,7 +145,7 @@ int does_int64_work()
   return 1;
 }
 main() {
-  exit(! does_int64_work());
+  return(! does_int64_work());
 }],
 [Ac_cachevar=yes],
 [Ac_cachevar=no],
@@ -201,7 +201,7 @@ int does_int64_snprintf_work()
   return 1;
 }
 main() {
-  exit(! does_int64_snprintf_work());
+  return(! does_int64_snprintf_work());
 }],
 [pgac_cv_snprintf_long_long_int_format=$pgac_format; break],
 [],

